### PR TITLE
ref: document `import-url` cloud versioning changes

### PR DESCRIPTION
**Pull request by pmrowla**
Per

- https://github.com/iterative/dvc/pull/8528 ?
- https://github.com/iterative/dvc/pull/8578
- https://github.com/iterative/dvc/pull/8581

### CODE REVIEW

File: content/docs/command-reference/import-url.md [hunk 1]
```diff
@@ -179,6 +179,12 @@ produces a regular stage in `dvc.yaml`.
 
 - `-h`, `--help` - prints the usage/help message, and exit.
 
+- `--version-aware` - capture cloud versioning information when importing the
+  file. By default, DVC will automatically capture cloud versioning information
+  if the URL contains a cloud versioning ID. When `--version-aware` is provided
+  along with a URL that does not contain a cloud versioning ID, DVC will capture
+  the latest version of the file.
```
Comment (dberenbaum) [inline content/docs/command-reference/import-url.md:186]: Let's also explain that dvc will pull that version from the source location even if it's overwritten, and will not push another copy of it to the remote.

cc @jorgeorpinel Is there somewhere in the data management user guide we want to this info also?

Comment (jorgeorpinel) [reply content/docs/command-reference/import-url.md:186]: We'll def. need UG updates to go over cloud versioning -- can't explain everything in an option text. For now I'd focus on what the flag does, and make some mentions/explanations in the cmd ref Description (which in this case is already super long and should be rewritten/ moved to UG eventually).

File: content/docs/user-guide/project-structure/dvc-files.md [hunk 2]
```diff
@@ -76,6 +76,7 @@ The following subfields may be present under `outs` entries:
...
 | `labels`                        | User-assigned labels to add to the data.                                                                                                                                                                                                                                                                                                              |
 | `meta`                          | Custom metadata about the data.                                                                                                                                                                                                                                                                                                                       |
+| `push`                          | (Optional) Whether the output file/dir should be pushed to the remote on `dvc push` (`true` by default: outputs are pushed to remotes). `push` only applies to <abbr>cached</abbr> outputs.                                                                                                                                                           |
```
Comment (jorgeorpinel) [inline content/docs/user-guide/project-structure/dvc-files.md:79]: Everything is optional except the first field. Also, slight rephrase (similar to `cache` text above):
```suggestion
| `push`                          | Whether or not this  file or directory, when previously <abbr>cached</abbr>, is uploaded to remote storage by `dvc push` (`true` by default). |
```

File: content/docs/user-guide/project-structure/dvcyaml-files.md [hunk 3]
```diff
@@ -265,6 +265,7 @@ These include a subset of the fields in `.dvc` file
 | `persist`    | Whether the output file/dir should remain in place during `dvc repro` (`false` by default: outputs are deleted when `dvc repro` starts)                                                                                                                                           |
 | `checkpoint` | (Optional) Set to `true` to let DVC know that this output is associated with [checkpoint experiments](/doc/user-guide/experiment-management/checkpoints). These outputs are reverted to their last cached version at `dvc exp run` and also `persist` during the stage execution. |
 | `desc`       | (Optional) User description for this output. This doesn't affect any DVC operations.                                                                                                                                                                                              |
+| `push`       | (Optional) Whether the output file/dir should be pushed to the remote on `dvc push` (`true` by default: outputs are pushed to remotes). `push` only applies to <abbr>cached</abbr> outputs.                                                                                       |
```
Comment (jorgeorpinel) [inline content/docs/user-guide/project-structure/dvcyaml-files.md:268]: Same
```suggestion
| `push`                          | Whether or not this  file or directory, when previously <abbr>cached</abbr>, is uploaded to remote storage by `dvc push` (`true` by default). |
```

**Review by jorgeorpinel — COMMENTED** [review]
Committing...

File: content/docs/user-guide/project-structure/dvcyaml-files.md [hunk 4]
```diff
@@ -265,6 +265,7 @@ These include a subset of the fields in `.dvc` file
 | `persist`    | Whether the output file/dir should remain in place during `dvc repro` (`false` by default: outputs are deleted when `dvc repro` starts)                                                                                                                                           |
 | `checkpoint` | (Optional) Set to `true` to let DVC know that this output is associated with [checkpoint experiments](/doc/user-guide/experiment-management/checkpoints). These outputs are reverted to their last cached version at `dvc exp run` and also `persist` during the stage execution. |
 | `desc`       | (Optional) User description for this output. This doesn't affect any DVC operations.                                                                                                                                                                                              |
+| `push`       | Whether or not this file or directory, when previously <abbr>cached</abbr>, is uploaded to remote storage by `dvc push` (`true` by default).                                                                                                                                      |
```
Comment (jorgeorpinel) [inline content/docs/user-guide/project-structure/dvcyaml-files.md:268]: > Echoing https://github.com/iterative/dvc/pull/8581#issuecomment-1336274557:

Should we plan to recommend this a lot in Data Pipeline docs? Specifically for intermediate pipeline outputs. Assuming the happy path out there is to push only raw data and final ML model files.

If we don't at least emphasize the possibility, users may realize too late they have pushed a bunch of intermediate output versions and they are pretty difficult to clean up with `dvc gc` ([support example](https://discord.com/channels/485586884165107732/1047778499999576084/1048151267190509600)).

File: content/docs/command-reference/import-url.md (hunk 1 above)
Comment (jorgeorpinel) [reply content/docs/command-reference/import-url.md:186]: p.s. Specifically, there's is some draft content about this in https://github.com/iterative/dvc.org/pull/4119/files#diff-d01612907e4ab14238625d537eaf42852d8566901d1bfe2fd3f2d4406a2d1dfc ATM.

**Comment by dberenbaum** [comment]
Related to #4089

File: content/docs/user-guide/project-structure/dvcyaml-files.md (hunk 4 above)
Comment (pmrowla) [reply content/docs/user-guide/project-structure/dvcyaml-files.md:268]: I'm not sure that not pushing is the right default behavior, even for intermediate outputs. If the user wants to take advantage of run-cache to not need to re-run stages that have already been reproduced, they still need to push/pull intermediate outs

File: content/docs/command-reference/update.md [hunk 5]
```diff
@@ -48,6 +48,10 @@ $ dvc update --rev master
   > Note that this changes the `rev` field in the import stage, fixing it to the
   > revision.
 
+  For stages created with `dvc import-url` and a cloud-versioned URL, `--rev`
+  can be used to specify a object version ID to use. By default, the import will
+  be updated to the latest version from cloud storage.
+
```
Comment (dberenbaum) [inline content/docs/command-reference/update.md:54]: Minor: link back to the `--version-aware` flag in `import-url`?

**Review by dberenbaum — APPROVED** [review]

File: content/docs/user-guide/project-structure/dvcyaml-files.md (hunk 4 above)
Comment (dberenbaum) [reply content/docs/user-guide/project-structure/dvcyaml-files.md:268]: @jorgeorpinel Thinking about it some more, I like the suggestion and think it makes sense as a possible product direction to make it easier to get started with pipelines, so let's brainstorm more on it.

**Comment by shcheklein** [comment]
@dberenbaum @pmrowla qq - what is the status of this, and the cloud versioning in general?

**Comment by dberenbaum** [comment]
> what is the status of this, and the cloud versioning in general?

It's on my plate to review and merge the outstanding docs PRs (here and https://github.com/iterative/dvc.org/pull/4165, and tracked more broadly in https://github.com/iterative/dvc.org/issues/4089). We discussed in sprint meeting and agreed we could merge now with an admon that it's experimental.

In Q1, the plan is to address performance issues and publicly promote it. Initial thoughts on blog post are in [Notion](https://www.notion.so/iterative/Blog-Post-DVC-Cloud-Versioning-65993772b6df4b62a2200096c501a368).

**Pull request merged**